### PR TITLE
CORTX:30573

### PR DESCRIPTION
**Pull request by ankita-93**
Signed-off-by: ankita-93 <ankita.thummar@seagate.com>

# Problem Statement
- TA under [https://jts.seagate.com/browse/CORTX-30573](url)

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/test_server_pod_failure.py [hunk 1]
```diff
@@ -505,3 +506,305 @@ def test_degraded_writes_safe_server_pod_shutdown(self):
...
+    @pytest.mark.ha
+    @pytest.mark.lc
+    @pytest.mark.tags("TEST-39908")
+    @CTFailOn(error_handler)
+    def test_degraded_deletes_safe_server_pod_shutdown(self):
+        """
+        Following test tests degraded deletes after safe server pod shutdown
+        """
+        LOGGER.info("STARTED: Test to verify degraded deletes after safe server pod shutdown.")
+        wr_bucket = HA_CFG["s3_bucket_data"]["no_buckets_for_deg_deletes"]      # 150 Buckets
```
Comment (ApurwaMohite) [inline tests/ha/test_server_pod_failure.py:686]: This comment is not required.

File: tests/ha/test_server_pod_failure.py [hunk 2]
```diff
@@ -505,3 +506,305 @@ def test_degraded_writes_safe_server_pod_shutdown(self):
...
+                                               passwd=S3_CFG["CliConfig"]["s3_account"]["password"])
+        assert_utils.assert_true(resp[0], resp[1])
+        access_key = resp[1]["access_key"]
+        secret_key = resp[1]["secret_key"]
+        self.test_prefix = 'test-39908'
+        self.s3_clean = {'s3_acc': {'accesskey': access_key, 'secretkey': secret_key,
+                                    'user_name': self.s3acc_name}}
+        s3_test_obj = S3TestLib(access_key=access_key, secret_key=secret_key,
+                                endpoint_url=S3_CFG["s3_url"])
+        LOGGER.info("Successfully created IAM account with name %s", self.s3acc_name)
```
Comment (ApurwaMohite) [inline tests/ha/test_server_pod_failure.py:705]: IAM user*

File: tests/ha/test_server_pod_failure.py [hunk 3]
```diff
@@ -505,3 +506,305 @@ def test_degraded_writes_safe_server_pod_shutdown(self):
...
+        """
+        LOGGER.info("STARTED: Test to verify degraded deletes after safe server pod shutdown.")
+        wr_bucket = HA_CFG["s3_bucket_data"]["no_buckets_for_deg_deletes"]      # 150 Buckets
+        del_bucket = wr_bucket - 10
+        event = threading.Event()
+        wr_output = Queue()
+        del_output = Queue()
+        LOGGER.info("Step 1: Create %s buckets & perform WRITEs with variable size objects.",
+                    wr_bucket)
+        LOGGER.info("Create IAM account with name %s", self.s3acc_name)
```
Comment (ApurwaMohite) [inline tests/ha/test_server_pod_failure.py:693]: IAM user*

**Review by sonalk0209 — APPROVED** [review]

**Review by ApurwaMohite — APPROVED** [review]

**Comment by ankita-93** [comment]
image
Collection log for this PR

**Pull request merged**